### PR TITLE
feat: Implement CLIENT GETREDIR

### PR DESCRIPTION
**Pull request by siyopao**
Part of #174.

### CODE REVIEW

File: tests/connection_test.ts [hunk 1]
```diff
@@ -77,6 +77,18 @@ suite.test("client setname & getname", async () => {
   assertEquals(await client.clientGetName(), "deno-redis");
 });
 
+suite.test("client getredir with no redirect", async () => {
+  assertEquals(await client.clientGetRedir(), -1);
+});
+
+suite.test("client getredir with redirect", async () => {
```
Comment (uki00a) [inline tests/connection_test.ts:84]: A small nitpick: please ensure that `tempClient` is closed!

```suggestion
suite.test("client getredir with redirect", async () => {
  const tempClient = await newClient({ hostname: "127.0.0.1", port: 7003 });
  try {
    const id = await tempClient.clientID();
    await client.clientTracking({ mode: "ON", redirect: id });
    assertEquals(await client.clientGetRedir(), id);
  } finally {
    tempClient.close();
  }
});
```

**Review by uki00a — CHANGES_REQUESTED** [review]
Hi @siyopao! Thank you for implementing this feature! :smiley:
I've left one comment. Please check it out!

File: tests/connection_test.ts (hunk 1 above)
Comment (siyopao) [reply tests/connection_test.ts:84]: Excellent point! Thank you. Fixed.

**Comment by siyopao** [comment]
Thanks for the review @uki00a!

**Review by uki00a — APPROVED** [review]
LGTM! Thank you for working on client side caching feature!

**Comment by siyopao** [comment]
It's my great pleasure. Thank you for the guidance @uki00a!

**Pull request merged**